streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -762,6 +763,21 @@ public static void addAvgAndSumMetricsToSensor(final Sensor sensor,
         );
     }
 
+    public static <R> R maybeMeasureLatency(final Supplier<R> action,
+                                            final Time time,
+                                            final Sensor sensor) {
+        if (sensor.shouldRecord()) {
+            final long startNs = time.nanoseconds();
+            try {
+                return action.get();
```

Comment:
Related to my comment above instead of a `Supplier`, maybe we could use take a `Runnable` here instead and remove the need for return values we'll never use.  Or if `Runnable` doesn't work, create a custom Functional interface with a void return type.  I recognize the return generic of `R`, but from looking at the code, we never expect a return value.

Reply:
One example where we need to return a value is in `Metered*Store` where we register the latency of getting data from the store. I could not yet use `StreamsMetricsImpl#maybeMeasureLatency()` there because I did not want to let the two PR depend from each other. However, I will provide two overloads of `maybeMeasureLatency()`, one with `Runnable` and one with `Supplier`.